agent/agent.go

```diff
@@ -1639,19 +1662,31 @@ func (a *agent) manageProcessPriority(ctx context.Context) ([]*agentproc.Process
 		// Getpriority actually returns priority for the nice value
 		// which is niceness + 20, so here 20 = a niceness of 0 (aka unset).
 		if score != 20 {
-			// We don't log here since it can get spammy
 			continue
 		}
 
-		err = proc.SetNiceness(a.syscaller, niceness)
-		if err != nil {
-			logger.Warn(ctx, "unable to set proc niceness",
-				slog.F("niceness", niceness),
-				slog.Error(err),
-			)
-			continue
+		if niceErr == nil {
+			err := proc.SetNiceness(a.syscaller, niceness)
+			if err != nil && !xerrors.Is(err, os.ErrPermission) {
+				logger.Warn(ctx, "unable to set proc niceness",
+					slog.F("niceness", niceness),
+					slog.Error(err),
+				)
```

Comment:
IDK if we should log these. If there's an issue preventing writing to this every time we try, it'll spam the logs. A customer already complained about a similar log spam.

At least it should be debounced and only logged once or once per minute or something.